### PR TITLE
Process Slashings for Newly Optimized Slasher

### PR DESCRIPTION
**Pull request by rauljordan**
This PR brings over logic to process slashings from feature/slasher into develop for a v2 release

### CODE REVIEW

File: beacon-chain/blockchain/testing/mock.go [hunk 1]
```diff
@@ -294,8 +294,8 @@ func (s *ChainService) ReceiveAttestationNoPubsub(context.Context, *ethpb.Attest
 	return nil
 }
 
-// AttestationPreState mocks AttestationPreState method in chain service.
-func (s *ChainService) AttestationPreState(_ context.Context, _ *ethpb.Attestation) (state.BeaconState, error) {
+// AttestationTargetState mocks AttestationTargetState method in chain service.
+func (s *ChainService) AttestationTargetState(ctx context.Context, target *ethpb.Checkpoint) (state.BeaconState, error) {
```
Comment (rkapka) [inline beacon-chain/blockchain/testing/mock.go:298]:
```suggestion
func (s *ChainService) AttestationTargetState(_ context.Context, _ *ethpb.Checkpoint) (state.BeaconState, error) {
```

File: beacon-chain/core/signing/signing_root.go [hunk 2]
```diff
@@ -96,6 +96,26 @@ func VerifySigningRoot(obj fssz.HashRoot, pub, signature, domain []byte) error {
 	return nil
 }
 
+// VerifyBlockHeaderSigningRoot verifies the signing root of a block header given it's public key, signature and domain.
```
Comment (rkapka) [inline beacon-chain/core/signing/signing_root.go:99]:
```suggestion
// VerifyBlockHeaderSigningRoot verifies the signing root of a block header given its public key, signature and domain.
```

File: beacon-chain/slasher/service.go [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/beacon-chain/operations/slashings"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/stategen"
 	"github.com/sirupsen/logrus"
 )
 
 var log = logrus.WithField("prefix", "slasher")
 
+type ServiceConfig struct {
```
Comment (rkapka) [inline beacon-chain/slasher/service.go:13]: Please add doc

File: beacon-chain/blockchain/testing/mock.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/blockchain/testing/mock.go:298]: I duplicated Deepsource on this one :-)

**Review by rkapka — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**